app/models/topic.rb

```diff
@@ -41,8 +41,13 @@ class Topic < ActiveRecord::Base
 
   acts_as_readable
 
-  def self.recent_topics
-    find_by_sql %q{
+  def self.recent_topics(user=nil)
+    if user && user.groups.any?
+      forumer_ids = user.groups.map(&:forumers).flatten.map(&:id)
+      constraint = "OR forumers.id IN (#{forumer_ids.join(",")})" if forumer_ids.any?
```

Comment:
Line is too long. [85/80]<br>Prefer single-quoted strings inside interpolations.